app/webpacker/controllers/products_controller.js

```diff
@@ -2,14 +2,19 @@ import ApplicationController from "./application_controller";
 
 export default class extends ApplicationController {
   static targets = ["loading"];
+  static values = { currentId: Number };
 
   connect() {
     super.connect();
     // Fetch the products on page load
     this.stimulate("Products#fetch");
   }
 
-  beforeReflex() {
+  beforeReflex(element) {
+    // To prevent the double click on the confirm modal's confirmation button
+    if (element.id === "confirmModalButton") {
+      window.dispatchEvent(new Event("modal:close"));
+    }
```

Comment:
Do we have a better way of disabling a button when it's clicked? That would prevent further interaction as well and can be used in many circumstances.

Reply:
This implements the **scenario#4** of the [issue](https://github.com/openfoodfoundation/openfoodnetwork/issues/11068). What do you think suggest?

Reply:
We can explore, however currently this implements the **scenario#4** of the [issue](https://github.com/openfoodfoundation/openfoodnetwork/issues/11068).

Reply:
I see. You are completely right following the spec. Sometimes, when another solution is a lot easier and doesn't affect the user experience much, we discuss alternative solutions. Nothing is set in stone.

I would keep your current solution though.